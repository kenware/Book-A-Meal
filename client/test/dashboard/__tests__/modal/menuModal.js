import React from 'react';
import { shallow } from 'enzyme';
import MenuModal from '../../../../components/dashboard/modals/menuModal';
import { props, emptyProps } from '../../dashboardMockData';

describe('Test menu component of dashboard', () => {
  it('renders correctly with empty props', () => {
    const tree = shallow(<MenuModal {...emptyProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with props', () => {
    const tree = shallow(<MenuModal {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
