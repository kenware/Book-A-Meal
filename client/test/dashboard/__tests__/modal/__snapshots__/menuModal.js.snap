// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test menu component of dashboard renders correctly with empty props 1`] = `
<div
  className=""
>
  <br />
  <div
    className="modal-header"
  >
    <h4
      className="justify l-r-pad-text"
    >
      Add 
       to cart 
      <br />
      Price: # 
      <p
        className="danger text-center"
      />
    </h4>
  </div>
  <div
    className="modal-contents"
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <img
      alt="meal"
      className="rounded-circle img-height"
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Quantity: 
    </span>
    <input
      className="modal-menu select"
      name="quantity"
      onChange={[MockFunction]}
      type="number"
    />
  </div>
  <div
    className="menu-footer"
  >
    <button
      className="modal-btn"
      onClick={[MockFunction]}
    >
      Add To Cart
    </button>
  </div>
</div>
`;

exports[`Test menu component of dashboard should render with props 1`] = `
<div
  className=""
>
  <br />
  <div
    className="modal-header"
  >
    <h4
      className="justify l-r-pad-text"
    >
      Add 
      rice
       to cart 
      <br />
      Price: # 
      <p
        className="danger text-center"
      />
    </h4>
  </div>
  <div
    className="modal-contents"
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <img
      alt="meal"
      className="rounded-circle img-height"
    />
  </div>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text"
    >
       Quantity: 
    </span>
    <input
      className="modal-menu select"
      name="quantity"
      onChange={[MockFunction]}
      type="number"
    />
  </div>
  <div
    className="menu-footer"
  >
    <button
      className="modal-btn"
      onClick={[MockFunction]}
    >
      Add To Cart
    </button>
  </div>
</div>
`;
